### ISSUE TITLE
[bug] Text in a shape doesn't export correctly

### ISSUE THREAD

**Issue by mcintyre94** [tldraw/tldraw#879]
TLDR file: [anchor-client-structure.tldr.txt](https://github.com/tldraw/tldraw/files/9259063/anchor-client-structure.tldr.txt) (added txt extension for github)

Screenshot in tldraw:
<img width="1165" alt="Screenshot 2022-08-04 at 11 58 57" src="https://user-images.githubusercontent.com/1711350/182831126-d28bdd72-13aa-4d48-a991-91dcde33b736.png">

Exported PNG (using Edit > Export As):
![anchor-client-structure-tldraw](https://user-images.githubusercontent.com/1711350/182831223-a40d4ba1-f17e-48e5-8a8e-cde8cc45523d.png)

- I get a similar result exporting to any of the image types. 
- I added the text by double clicking the shapes and typing it. If I instead use the separate text tool and manually place the text within the shapes, it exports correctly
- I couldn't find a simpler repro sorry! It behaves correctly when I try simpler cases.

**Comment by mcintyre94** [comment]
Another example which I think is the same issue:

[pixel-map.tldr.txt](https://github.com/tldraw/tldraw/files/9269350/pixel-map.tldr.txt)

Screenshot:
<img width="1609" alt="Screenshot 2022-08-05 at 14 56 18" src="https://user-images.githubusercontent.com/1711350/183094762-d45d5628-2bf4-46a3-8822-b2ca86511cb0.png">

Exported:
![Page 1](https://user-images.githubusercontent.com/1711350/183094836-d2089990-3033-48c7-a283-33d5681a7e60.png)

**Comment by mcndt** [comment]
It seems like this regressed recently, see my report in #913.
